orchestrator/cli/migrate_domain_models.py

```diff
@@ -0,0 +1,346 @@
+
+    Args:
+        - model_class: a product class (SubscriptionModel) or product block class (ProductBlockModel).
+
+    Returns a Dict with product block models by product block name.
+    """
+    product_blocks_types_in_model = get_depends_on_product_block_type_list(
+        model_class._get_depends_on_product_block_types()
+    )
+
+    new_blocks = {
+        block.name: block for block in product_blocks_types_in_model if block.name not in product_blocks.keys()
+    }
+    product_blocks = {**product_blocks, **new_blocks}
+
+    blocks_map = {
+        name: block_cls
+        for block in new_blocks.values()
+        for name, block_cls in map_product_blocks_in_class(block, product_blocks).items()
+        if block_cls.name not in product_blocks.keys()
+    }
+    return {**product_blocks, **blocks_map}
+
+
+def map_product_blocks(product_classes: List[Type[SubscriptionModel]]) -> Dict[str, Type[ProductBlockModel]]:
+    """Create mapping of all existing product block models related to products.
+
+    Args:
+        - product_classes: List of product classes.
+
+    Returns a Dict with product block models by product block name.
+    """
+
+    product_blocks: Dict[str, Type[ProductBlockModel]] = {}
+    for product_class in product_classes:
+        product_blocks = {**product_blocks, **map_product_blocks_in_class(product_class, product_blocks)}
+    return product_blocks
+
+
+def map_differences_unique(
+    registered_products: Dict[str, Type[SubscriptionModel]], existing_products: List[Tuple[str, UUID]]
+) -> Dict[str, Dict[str, Dict[str, Set[str]]]]:
+    """Create a unique map for products and product block differences from the database.
+
+    Args:
+        - existing_products: List with tuples of product name and its uuid that exist in the database.
+
+    Returns a dict with product and product block differences from the database without duplicates.
+    """
+    model_diffs: Dict[str, Dict[str, Dict[str, Set[str]]]] = {"products": {}, "blocks": {}}
+
+    for product_name, product_id in existing_products:
+        if should_skip(product_name) or product_name not in registered_products.keys():
+            continue
+
+        product_class = registered_products[product_name]
+        if diff := product_class.diff_product_in_database(product_id):
+            model_diffs["products"][product_name] = {**diff[product_name], "missing_in_depends_on_blocks": []}  # type: ignore
+
+            if missing_in_depends_on_blocks := diff[product_name].get("missing_in_depends_on_blocks"):
+                model_diff_keys = model_diffs["blocks"].keys()
+                # since missing_in_depends_on_blocks is always a dict the type is ignored.
+                new_model_diffs = {
+                    name: diff for name, diff in missing_in_depends_on_blocks.items() if name not in model_diff_keys  # type: ignore
+                }
+                model_diffs["blocks"] = {**model_diffs["blocks"], **new_model_diffs}  # type: ignore
```

Comment:
The for looks a bit messy: maybe goot to lift it into a separate function or comprehension.